python/ray/_raylet.pyx

```diff
@@ -0,0 +1,323 @@
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+from libc.stdint cimport int32_t, int64_t
+from libcpp cimport bool as c_bool
+from libcpp.memory cimport unique_ptr
+from libcpp.string cimport string as c_string
+from libcpp.utility cimport pair
+from libcpp.unordered_map cimport unordered_map
+from libcpp.vector cimport vector as c_vector
+
+from ray.includes.common cimport (
+    CUniqueID, CTaskID, CObjectID, CFunctionID, CActorClassID, CActorID,
+    CActorHandleID, CWorkerID, CDriverID, CConfigID, CClientID,
+    CLanguage, CRayStatus, LANGUAGE_CPP, LANGUAGE_JAVA, LANGUAGE_PYTHON)
+from ray.includes.libraylet cimport (
+    CRayletClient, GCSProfileTableDataT, GCSProfileEventT,
+    ResourceMappingType, WaitResultPair)
+from ray.includes.task cimport CTaskSpecification
+from ray.includes.ray_config cimport RayConfig
+from ray.utils import decode
+
+from cython.operator import dereference, postincrement
+cimport cpython
+
+include "includes/unique_ids.pxi"
+include "includes/ray_config.pxi"
+include "includes/task.pxi"
+
+
+if cpython.PY_MAJOR_VERSION >= 3:
+    import pickle
+else:
+    import cPickle as pickle
+import numpy
+
+
+cdef int check_status(const CRayStatus& status) nogil except -1:
+    if status.ok():
+        return 0
+
+    with gil:
+        message = status.message().decode()
+        raise Exception(message)
+
+
+cdef c_vector[CObjectID] ObjectIDsToVector(object_ids):
+    """A helper function that converts a Python list of object IDs to a vector.
+
+    Args:
+        object_ids (list): The Python list of object IDs.
+
+    Returns:
+        The output vector.
+    """
+    cdef:
+        ObjectID object_id
+        c_vector[CObjectID] result
+    for object_id in object_ids:
+        result.push_back(object_id.data)
+    return result
+
+
+cdef VectorToObjectIDs(c_vector[CObjectID] object_ids):
+    result = []
+    for i in range(object_ids.size()):
+        result.append(ObjectID.from_native(object_ids[i]))
+    return result
+
+
+def compute_put_id(TaskID task_id, int64_t put_index):
+    if put_index < 1 or put_index > kMaxTaskPuts:
+        raise ValueError("The range of 'put_index' should be [1, %d]" % kMaxTaskPuts)
+    return ObjectID.from_native(ComputePutId(task_id.data, put_index))
+
+
+def compute_task_id(ObjectID object_id):
+    return TaskID.from_native(ComputeTaskId(object_id.data))
+
+
+cdef c_bool is_simple_value(value, int *num_elements_contained):
+    # Because "RayConfig" have a private destructor,
```

Comment:
```suggestion
    # Because "RayConfig" has private destructor,
```

Reply:
though i'm not sure how necessary this comment is. This is just the intended usage of `RayConfig` since it's a singleton.

Reply:
I removed it